packages/devextreme/js/__internal/ui/list/m_list.edit.decorator.reorder.ts

```diff
@@ -59,43 +74,56 @@ class EditDecoratorReorder extends EditDecorator {
     return result;
   }
 
-  _dragStartHandler(e) {
+  _dragStartHandler(e: DragStartEvent): void {
     if (this._lockedDrag) {
       e.cancel = true;
     }
   }
 
-  _dragChangeHandler(e) {
-    if (this._groupedEnabled && !this._sameParent(e.fromIndex, e.toIndex)) {
+  _dragChangeHandler(e: DragChangeEvent): void {
+    if (
+      this._groupedEnabled
+      && e.fromIndex
+      && e.toIndex
+      && !this._sameParent(e.fromIndex, e.toIndex)
+    ) {
       e.cancel = true;
     }
   }
 
-  _sameParent(fromIndex, toIndex) {
+  _sameParent(fromIndex: number, toIndex: number): boolean {
     const $dragging = this._list.getItemElementByFlatIndex(fromIndex);
     const $over = this._list.getItemElementByFlatIndex(toIndex);
 
     return $over.parent().get(0) === $dragging.parent().get(0);
   }
 
-  _reorderHandler(e) {
+  _reorderHandler(e: ReorderEvent): void {
     const $targetElement = this._list.getItemElementByFlatIndex(e.toIndex);
-    this._list.reorderItem($(e.itemElement), $targetElement);
+    this._list.reorderItem(e.itemElement, $targetElement.get(0));
   }
 
-  afterBag(config?) {
+  afterBag(config?: BagConfig): void {
     const $handle = $('<div>').addClass(REORDER_HANDLE_CLASS);
 
-    eventsEngine.on($handle, 'dxpointerdown', (e) => {
-      this._lockedDrag = !isMouseEvent(e);
-    });
-    eventsEngine.on($handle, 'dxhold', { timeout: 30 }, (e) => {
-      e.cancel = true;
-      this._lockedDrag = false;
-    });
-
-    config.$container
-      .addClass(REORDER_HANDLE_CONTAINER_CLASS)
+    eventsEngine.on(
+      $handle,
+      pointerEvents.down,
+      (e: DxEvent<PointerEvent | MouseEvent | TouchEvent>): void => {
+        this._lockedDrag = !isMouseEvent(e);
+      },
+    );
+    eventsEngine.on(
+      $handle,
+      holdEvent.name,
+      { timeout: 30 },
+      (e: ItemHoldEvent & Cancelable): void => {
```

Comment:
Here is usual event subscription, I suppose DxEvent<> & Cancellable expects here